main.tf

```diff
@@ -52,7 +52,7 @@ resource "aws_iam_role_policy_attachment" "lambda_insights" {
 
 # S3 object to hold the deployed artifact
 resource "aws_s3_bucket_object" "lambda_deploy_object" {
-  for_each    = var.image_uri == null ? [1] : []
+  for_each    = var.image_uri == null ? { 1 = 1 } : {}
```

Comment:
@ianbender let me know if you have any thoughts on the error related to this, I'm suspicious that there's an underlying issue that this doesn't fix because there are many other places that use `[1] : []` without issue... https://github.com/highwingio/data-dags/actions/runs/3101006002/jobs/5021978070#step:3:1997

Reply:
@nmnielsen you can use `count` instead if you just want this to act like a conditional resource.
```tf
  count = var.image_uri == null ? 1 : 0
```

Reply:
aha, excellent, much cleaner!